qtconsole/tests/test_console_widget.py

```diff
@@ -415,35 +417,140 @@ def is_complete(self, source):
         w.execute(interactive=True)
         assert responses == [('complete', None)]
 
-    def test_debug(self):
-        """
-        Make sure the cursor works while debugging 
-        
-        It might not because the console is "_executing"
-        """
-        # Kernel client to test the responses of is_complete
-        class TestIPyKernelClient(object):
-            def is_complete(self, source):
-                tm = TransformerManager()
-                check_complete = tm.check_complete(source)
-                responses.append(check_complete)
 
-        # Initialize widget
-        responses = []
-        w = ConsoleWidget()
-        w._append_plain_text('Header\n')
-        w._prompt = 'prompt>'
-        w._show_prompt()
-        w.kernel_client = TestIPyKernelClient()
-        control = w._control
+@pytest.fixture
+def qtconsole():
+    """Qtconsole fixture."""
+    # Create a console with the ipdb kernel
+    console = JupyterQtConsoleApp()
+    console.initialize()
+
+    console.window.confirm_exit = False
+    console.window.show()
+    return console
+
+
+def test_scroll(qtconsole):
+    """
+    Make sure the scrolling works.
+    """
+    window = qtconsole.window
+    shell = window.active_frontend
+    wait_loop = QtCore.QEventLoop()
+    shell.executed.connect(wait_loop.quit)
+    control = shell._control
+    scroll_bar = control.verticalScrollBar()
```

Comment:
The previous six lines are completely unnecessary by using `qtbot`. Please remove them and replace them with the code I used in the referenced file.